A README.md for the tools (#3)
MMTIS/badger: tools/README.md

# Tools for badger
by Matthias Günter

In this directory there is a set of tools that can be used with the badger conversions:
* a script runner
* a gtfs validator
* a gtfs visualisation
* an xml validator (used to validate the NeTEx with the relevant XSD)

Future tools will contain:
* a relation explorer in NeTEx
* a SIRI ET/PT export of a given operating day
* some assertions for NeTex
* some statistics for NeTEx

We might also add tools, that correct common problems in certain data sets.

## The XML (NeTEx) validator
in the `tools` folder there is the a tool for validating xml files.
```

> uv run python -m tools.xml.validator path_to_xml_file.xml.zip  path_to_main_xsd_file

Reviewer: ```suggestion
uv run python -m tools.xml.validator path_to_xml_file.xml.zip path_to_main_xsd_file
```

```

> `path_to_main_xsd_file` is in `../schema/netex/xsd/netex_publication.xsd` if the project is installed 
> regularly.

Reviewer: ```suggestion
`path_to_main_xsd_file` is in `../schema/netex/xsd/netex_publication.xsd` if the project is installed regularly.
```

`path_to_xml_file.xml.zip` can be an xml, a zip or a gz file.

## The GTFS validator
in the `tools` folder there is the a tool for validating GTFS files. It uses the Mobility Data validator: https://github.com/MobilityData/gtfs-validator/#readme

> You need to install this locally, with an acceptabler java run time environment.

Reviewer: ```suggestion
You need to install this locally, with an accepted Java runtime environment.
```
Specify minimum version here too.

> The current project expects it in the tools-folder. You can change the name and place in the configuration: 

Reviewer: ```suggestion
The current project expects it in the tools-folder.
You can change the name and place in the configuration: 
```

```
gtfs_validator = "./gtfs-validator-7.0.0-cli.jar"  # location of the gtfs validator.jar.
```
You can run it as
```
uv run python -m tools.gtfs_validator path_to_gtfs_file.zip  path_to_report_file
```

> The script calls the jar file to do it works. The result contains two json-files and an html report.

Reviewer: ```suggestion
The script calls the jar file to do it works.
The result contains two json-files and an html report.
```

If there is no error found, then the excecution is considered successful.

## The GTFS Visualisation

> In the `tools` folder there is the a tool for GTFS visualisation. The idea is to see the geographical extension of 

Reviewer: ```suggestion
In the `tools` folder there is the a tool for GTFS visualisation.
The idea is to see the geographical extension of 
```

> In the `tools` folder there is the a tool for GTFS visualisation. The idea is to see the geographical extension of 
> transport networks. For this reasons "some" stops and "some" routes are shown on a map

Reviewer: ```suggestion
In the `tools` folder there is the a tool for GTFS visualisation.
The idea is to see the geographical extension of transport networks.
For this reasons "some" stops and "some" routes are shown on a map
```

```
uv run python -m tools.gtfs_map_visualisation path_to_gtfs_file.zip  path_to_folium_map.html one_of_n_routes
```

> `one_of_n_routes` is a number. If set to 1 every route is shown, if set to 10, then only one in 10 routes and one in ten
> stops is shown. This is nonsensical for processing, but it shows the geographical extension. Set to 1 one has
> a reasonable view on the "lines" within the GTFS file.

Reviewer: ```suggestion
`one_of_n_routes` is a number.
If set to 1 every route is shown, if set to 10, then only one in 10 routes and one in ten stops is shown.
This is nonsensical for processing, but it shows the geographical extension.
Set to 1 one has a reasonable view on the "lines" within the GTFS file.
```

> On startup in the report only the stops are shown. To see the trips check the box in the top right corner.

Reviewer: ```suggestion
On startup in the report only the stops are shown.
To see the trips check the box in the top right corner.
```

## The script runner

> in the `tools` folder there is the a tool for running scripts to process everything

Reviewer: ```suggestion
In the `tools` folder there is the a tool for running scripts to process everything.
```

```
uv run python -m tools.script_runner path_to_script_file.txt  name_of_logfile.log name_of_block (--begin_step=no_of_step) (--this_step=no_of_step) (--end_step=no_of_step)
```
There are predefined script files already in the `tools/tools_script` folder

> `no_of_step` is the sequence number of the step (!are counted in the script and not using the "step-no)

Reviewer: Review this line.

Author: done

`name_of_block` the identier of the block. `all` means all blocks (are processed sequentially.)

### Basic idea

> the script runner allows to run scripts.
> In the script file there are blocks that can be accessed

Reviewer: ```suggestion
The script runner allows to run scripts.
In the script file there are blocks that can be executed.
```
